api/src/main/java/io/smallrye/stork/integration/DefaultStorkInfrastructure.java

```diff
@@ -0,0 +1,15 @@
+package io.smallrye.stork.integration;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Supplier;
+
+public class DefaultStorkInfrastructure implements StorkInfrastructure {
+    private final Map<Class<?>, Object> utilities = new ConcurrentHashMap<>();
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public <T> T get(Class<T> utilityClass, Supplier<T> defaultSupplier) {
+        return (T) utilities.computeIfAbsent(utilityClass, key -> defaultSupplier.get());
```

Comment:
Probability check that utilityClass and defaultSupplier are not null.

Reply:
done